src/modm/platform/adc/stm32f0/adc_impl.hpp.in

```diff
@@ -360,3 +367,127 @@ modm::platform::Adc{{ id }}::setWaitMode(const bool enable)
 		ADC1->CFGR1 &= ~ADC_CFGR1_WAIT;
 	}
 }
+
+%% if target.family in ["g0"]
+uintptr_t
+modm::platform::Adc{{ id }}::getDataRegisterAddress()
+{
+	return (uintptr_t) &(ADC1->DR);
+}
+
+void
+modm::platform::Adc{{ id }}::enableRegularConversionExternalTrigger(
+	ExternalTriggerPolarity externalTriggerPolarity,
+	RegularConversionExternalTrigger regularConversionExternalTrigger)
+{
+	const auto polarity =
+		(static_cast<uint32_t>(externalTriggerPolarity) << ADC_CFGR1_EXTEN_Pos);
+	const auto externalTrigger =
+		(static_cast<uint32_t>(regularConversionExternalTrigger) << ADC_CFGR1_EXTSEL_Pos);
+	const auto mask = ADC_CFGR1_EXTEN_Msk | ADC_CFGR1_EXTSEL_Msk;
+	ADC1->CFGR1 = (ADC1->CFGR1 & ~mask) | polarity | externalTrigger;
+}
+
+void
+modm::platform::Adc{{ id }}::enableDmaMode()
+{
+	ADC1->CFGR1 |= ADC_CFGR1_DMAEN;
+}
+
+void
+modm::platform::Adc{{ id }}::disableDmaMode()
+{
+	ADC1->CFGR1 &= ~ADC_CFGR1_DMAEN;
+}
+
+bool
+modm::platform::Adc{{ id }}::getAdcEnabled()
+{
+	return (ADC1->CR & ADC_CR_ADEN);
+}
+
+
+void
+modm::platform::Adc{{ id }}::enableDmaRequests()
+{
+	ADC1->CFGR1 |= ADC_CFGR1_DMACFG;
+}
+
+void
+modm::platform::Adc{{ id }}::disableDmaRequests()
+{
+	ADC1->CFGR1 &= ~ADC_CFGR1_DMACFG;
+}
+
+void
+modm::platform::Adc{{ id }}::waitChannelConfigReady()
+{
+	while (!(ADC1->ISR & ADC_ISR_CCRDY));
+}
+
+void
+modm::platform::Adc{{ id }}::resetChannelConfigReady(){ADC1->ISR &= ~ADC_ISR_CCRDY;}
```

Comment:
This is not correct according to the reference manual. The bit is cleared by writing 1 not 0.